Changed access permissions for CouncilDetails and ClubDetails endpoint

Vikhyath08 opened the pull request:
Added a new Permission class (`AllowCouncilHead`) which only allows Council Heads to access methods which are not `SAFE_METHODS`.

krashish8 commented:
@Vikhyath08 I think it would be better to replace `permissions.AllowAny` with `AllowCouncilHead`, instead of adding the extra permission with AllowAny permission. It works good, however!

Also, there is a similar permission issue with the ClubDetails endpoint, so you can make the changes there too.

krashish8 commented:
@nishantwrp Are the CircleCI lint-checks disabled?

Vikhyath08 commented:
Oof. Yes.
No idea what I was thinking there.

Yes, I'll make the changes in the ClubDetails as well

Vikhyath08 commented:
Added permission for the ClubDetailView endpoint, similar to the CouncilDetailView permission.
Also fixed an issue with the previous PR that a council head was able to edit the details of other councils as well.

nishantwrp commented:
@Vikhyath08 merge this branch from the latest `master`. So, the lint checks can run.

krashish8 commented on workshop/permissions.py:
```diff
@@ -76,3 +77,33 @@ def has_object_permission(self, request, view, obj):
+        # pylint: disable=no-member
+        profile = UserProfile.objects.get(user=request.user)
+        club = obj
+        if club in profile.get_club_privileges:
```
```suggestion
        if club in profile.get_club_privileges():
```

krashish8 commented on workshop/permissions.py:
```diff
@@ -76,3 +77,33 @@ def has_object_permission(self, request, view, obj):
+        # pylint: disable=no-member
+        profile = UserProfile.objects.get(user=request.user)
+        council = obj
+        if council in profile.get_council_privileges:
```
```suggestion
        if council in profile.get_council_privileges():
```

krashish8 commented on workshop/permissions.py:
```diff
@@ -76,3 +77,33 @@ def has_object_permission(self, request, view, obj):
+        council = obj
+        if council in profile.get_council_privileges:
+            return True
+        return False
```
```suggestion
        return False

```
Please merge this branch with the latest master, so that lint checks can run, and fix the lint check errors (if any).

krashish8 commented on workshop/views.py:
```diff
@@ -31,7 +31,7 @@ class ClubDetailView(generics.RetrieveUpdateAPIView):
     # pylint: disable=no-member
     queryset = Club.objects.all()
-    permission_classes = (permissions.AllowAny,)
+    permission_classes = (AllowParticularClubHead)
```
```suggestion
    permission_classes = (AllowParticularClubHead,)
```

krashish8 commented on workshop/views.py:
```diff
@@ -116,7 +116,7 @@ class CouncilDetailView(generics.RetrieveUpdateAPIView):
     # pylint: disable=no-member
     queryset = Council.objects.all()
-    permission_classes = (permissions.AllowAny,)
+    permission_classes = (AllowParticularCouncilHead)
```
```suggestion
    permission_classes = (AllowParticularCouncilHead,)
```
A tuple with a single element should have a comma at the end, otherwise it will give `500 Error` on accessing the endpoint.

krashish8 reviewed (changes requested):
Please make sure to test the endpoints thoroughly before making the PR, so that you know that your objective is fulfilled, and your change doesn't break anything else.

krashish8 commented:
@Vikhyath08 Everything looks good. Just merge your branch with the latest master one last time, so that lint checks would run, and fix the lint-check errors if any.

Vikhyath08 reviewed (commented):
Committed changes

krashish8 commented:
@Vikhyath08 I mean, the master branch of this repository has updated, to include the files for running lint checks. So, merge your branch with the updated master branch, so that link checks will run. We can merge this PR only when all the checks pass.

In case you don't know how to merge with the updated master branch, follow these steps:
```
git checkout modify-council-permissions    # Ensure that you are in this branch
git remote add upstream https://github.com/IIT-BHU-InstiApp/workshops-app-backend
git fetch upstream
git merge upstream/master
git push origin modify-council-permissions
```

Also, fix the lint check errors, if any.

krashish8 reviewed (approved):
LGTM! 🤘🏻
